agent/commands/run.go

```diff
@@ -94,37 +83,55 @@ func Run() {
 	}()
 	go func() {
 		defer wg.Done()
-		localServer.Run(ctx)
+		localServer.Run(ctx, reloadCh)
 		cancel()
 	}()
 
-	for {
-		_, err = configStorage.Reload(l)
-		if err != nil {
-			l.Fatalf("Failed to load configuration: %s.", err)
-		}
+	clientCtx, cancelClientCtx := context.WithCancel(ctx)
 
-		cfg := configStorage.Get()
+	_ = client.Run(clientCtx)
+	cancelClientCtx()
 
-		config.ConfigureLogger(cfg)
-		logStore.Resize(cfg.LogLinesCount)
-		l.Debugf("Loaded configuration: %+v", cfg)
+	wg.Wait()
 
-		logrus.Infof("Window check connection time is %.2f hour(s)", cfg.WindowConnectedTime.Hours())
-		connectionUptimeService.SetWindowPeriod(cfg.WindowConnectedTime)
+	if ctx.Err() != nil {
+		l.Info(ctx.Err())
+	}
 
-		clientCtx, cancelClientCtx := context.WithCancel(ctx)
+	if x := <-reloadCh; x {
+		close(reloadCh)
+		run(l, logStore)
```

Comment:
Recursion isn't a good idea here.

Reply:
@BupycHuk Please take a look if "for" solution is better.